deployments/charts/kuma/README.md

```diff
@@ -38,23 +43,32 @@ A Helm chart for the Kuma Control Plane
+| controlPlane.lifecycle | object | `{}` | Pod lifecycle settings (useful for adding a preStop hook, when using AWS ALB or NLB) |
+| controlPlane.terminationGracePeriodSeconds | int | `30` | Number of seconds to wait before force killing the pod. Make sure to update this if you add a preStop hook. |
 | controlPlane.tls.general.secretName | string | `""` | Secret that contains tls.crt, tls.key [and ca.crt when no controlPlane.tls.general.caSecretName specified] for protecting Kuma in-cluster communication |
 | controlPlane.tls.general.caSecretName | string | `""` | Secret that contains ca.crt that was used to sign cert for protecting Kuma in-cluster communication (ca.crt present in this secret have precedence over the one provided in the controlPlane.tls.general.secretName) |
 | controlPlane.tls.general.caBundle | string | `""` | Base64 encoded CA certificate (the same as in controlPlane.tls.general.secret#ca.crt) |
 | controlPlane.tls.apiServer.secretName | string | `""` | Secret that contains tls.crt, tls.key for protecting Kuma API on HTTPS |
 | controlPlane.tls.apiServer.clientCertsSecretName | string | `""` | Secret that contains list of .pem certificates that can access admin endpoints of Kuma API on HTTPS |
-| controlPlane.tls.kdsGlobalServer.secretName | string | `""` | Secret that contains tls.crt, tls.key for protecting cross cluster communication |
-| controlPlane.tls.kdsZoneClient.secretName | string | `""` | Secret that contains ca.crt which was used to sign KDS Global server. Used for CP verification |
+| controlPlane.tls.kdsGlobalServer.secretName | string | `""` | Name of the K8s TLS Secret resource. If you set this and don't set create=true, you have to create the secret manually. |
+| controlPlane.tls.kdsGlobalServer.create | bool | `false` | Whether or not to create the TLS secret in helm. |
+| controlPlane.tls.kdsGlobalServer.cert | string | `""` | The TLS certificate to offer. |
+| controlPlane.tls.kdsGlobalServer.key | string | `""` | The TLS key to use. |
+| controlPlane.tls.kdsZoneClient.secretName | string | `""` | Name of the K8s TLS Secret resource. If you set this and don't set create=true, you have to create the secret manually. |
+| controlPlane.tls.kdsZoneClient.create | bool | `false` | Whether or not to create the TLS secret in helm. |
+| controlPlane.tls.kdsZoneClient.cert | string | `""` | The TLS certificate to expect. |
 | controlPlane.image.pullPolicy | string | `"IfNotPresent"` | Kuma CP ImagePullPolicy |
 | controlPlane.image.repository | string | `"kuma-cp"` | Kuma CP image repository |
 | controlPlane.image.tag | string | `nil` | Kuma CP Image tag. When not specified, the value is copied from global.tag |
+| controlPlane.additionalK8sSecrets | list | `[]` | list of additional secret resources to create; for each specify the name and stringData name is rendered as a template using (tpl $) |
```

Comment:
So those secrets are not mounted to the CP?
Do they need to have CP labels (I'm thinking about moving it to root)?

Can you describe a practical example of this?

I'm wondering if there is a better pattern to solve this with HELM. Have you considered creating a simple chart that has a dependency on Kuma and includes your additional resources?